lxd/instance/drivers/driver_qemu.go

```diff
@@ -921,16 +927,45 @@ func (d *qemu) Start(stateful bool) error {
 		"-readconfig", confFile,
 		"-pidfile", d.pidFilePath(),
 		"-D", d.LogFilePath(),
-		"-chroot", d.Path(),
+	}
+
+	// If stateful, restore now.
+	if stateful {
+		if !d.stateful {
+			err = fmt.Errorf("Instance has no existing state to restore")
+			op.Done(err)
+			return err
+		}
+
+		qemuCmd = append(qemuCmd, "-incoming", "defer")
+	} else {
+		// The chroot option can't be used when restoring a migration stream.
+		qemuCmd = append(qemuCmd, "-chroot", d.Path())
+
+		if d.stateful {
+			// Stateless start requested but state is present, delete it.
+			err := os.Remove(d.StatePath())
```

Comment:
@stgraber is there a possibility of having an `hasState()` function that can inspect the contents of the StatePath and avoid needing to maintain the UpdateInstanceStatefulFlag in the DB? I'm wondering why we have two ways to indicate if an instance has state?

Reply:
There is a Stateful property in api.Instance and api.InstanceSnapshot so we need to be able to fill that data from the database alone or we'd need to mount every instance and snapshot whenever you run 'lxc list'.

Reply:
makes sense thanks